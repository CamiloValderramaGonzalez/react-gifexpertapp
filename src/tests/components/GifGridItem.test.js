import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en GifGrid', () => {
    const title = 'titulo';
    const url = 'http://url.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener el src y el alt', () => {
        const img = wrapper.find('img');

        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);

    })

    test('Debe tener la clase fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.props().className;

        expect(className.includes('animate__fadeIn')).toBe(true);

    })

})

