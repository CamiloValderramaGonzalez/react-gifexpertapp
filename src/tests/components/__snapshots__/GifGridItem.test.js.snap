// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="http://url.com"
  />
  <p>
    titulo
  </p>
</div>
`;
